### ISSUE TITLE
Does this node work with the adafruit bno055 board?

### ISSUE THREAD

**Issue by jack-digilabs** [yoshito-n-students/bno055_usb_stick#13]
This looks like a great ROS package however I don't have a bno055 USB stick yet (in the mail) but I do have the adafruit bno055 board connected via a serial cable. Does the this ROS package work with the adafruit bno055?

I think the adafruit bno055 and the bno055 stick use the same serial protocol so I'm not sure why this isn't working. I'm getting the error "handleWaitResponse: End of file" whenever I try launching a bno055_usb_stick_node. The error prints off about every second. 

I am currently using the ROS package rr_bosch_bno055_imu to get data from the adafruit bno055 board and it works fine however it doesn't give access to all the data from the bno055 and it looks cumbersome to modify so that's why I'm hoping this one can work.

Any info would be greatly appreciated.

I'm running ROS kinetic on ubuntu 16.04.

**Comment by yoshito-okada** [comment]
I have never tested this package with the adafruit board.  It might be helpful for you to compare codes of rr_bosch_bno055_imu and my package.

**Comment by jack-digilabs** [comment]
Based on your answer to issue #5 and the constants.hpp file, it doesn't look like it's using the bno055 UART interface protocol. For example. a read command in the documentation is supposed to be only 4 bytes long however the commands stored in the constants.hpp are much longer. 

Did you try the bno055 protocol to start with and found it didn't work? @BillWSY did you try the given bno055 protocol at all or just used this package?

Attached is the bno055 UART interface spec that works on the adafruit boards: 
[BST-BNO055-AN012.pdf](https://github.com/yoshito-n-students/bno055_usb_stick/files/2837630/BST-BNO055-AN012.pdf)

**Comment by BillWSY** [comment]
It had been too long, but before using this USB stick I was using an Adafruit board with a ROS package called `bosch_imu_driver`. Google gives multiple results now so I don't know which I did use.

I think the USB stick has a special protocol -- you can use something like an interceptor to analyze the protocol of the USB stick.

**Comment by jack-digilabs** [comment]
Thanks for the response! I'm currently using a fork of the bosch_imu_driver so I'm familiar with it and it mostly works except for the quaternions from what I can tell.

I don't have a bno055 USB stick yet so I can't analyze the protocol yet but when it get's here I might try. I'll probably be bouncing between them so if I could use one ROS driver for both that'd be ideal.
I submitted a contact form to Bosch-Sensortec asking about the protocol but haven't heard anything back yet.

**Comment by jack-digilabs** [comment]
I heard back from Bosch-Sensortec and they said the protocol is proprietary so they weren't immediately going to give it to me, however looking into the USB stick a bit more it shows the Bosch IMU connected to an Atmel chip and then the Atmel chip communicates over USB. I'm pretty sure they just used the code from the BNO055 Xplained Pro Extension Kit that connects to the Atmel’s Xplained board and used the code they already had written for it. Consequently, the protocol to talk to the USB stick is the serial protocol Atmel uses for their Xplained board and not the Bosch bno055 protocol.

Basically, the bno055 USB stick does not use the bno055 protocol and so is not directly compatible.

I've tried the USB stick now with @yoshito-okada ROS driver and have had no issues and am able to get the full 100Hz rate compared to only 55hz-ish with the python ROS driver for the adafruit.

Thanks again @yoshito-okada for for the awesome code!

**Comment by yoshito-okada** [comment]
Thank you for asking to Bosch and making it clear.  I would like to close this thread.
